test/embedded-refract-test.js

```diff
@@ -0,0 +1,442 @@
+var _ = require('lodash');
+var expect = require('./spec-helper').expect;
+var minim = require('../lib/minim');
+
+describe('Embedded Refract', function() {
+  context('when converting from embedded refract', function() {
+    context('when given a normal object', function() {
+      var obj = {
+        foo: 'bar'
+      };
+
+      var element;
+
+      before(function() {
+        element = minim.convertFromEmbedded(obj);
+      });
+
+      it('returns the correct value', function() {
+        expect(element.toValue()).to.deep.equal(obj);
+      });
+    });
+
+    context('when given embedded refract object', function() {
+      // An object { b: 2 } with embedded refract
+      var obj = {
+        refract: {
+          element: 'object',
+          meta: {
+            id: 'test-id'
+          },
+          attributes: {
+            a: 1
+          }
+        },
+        b: 2
+      }
+
+      var refract = {
+        element: 'object',
+        meta: {
+          id: 'test-id'
+        },
+        attributes: {
+          a: 1
+        },
+        content: [
+          {
+            element: 'member',
+            meta: {},
+            attributes: {},
+            content: {
+              key: {
+                element: 'string',
+                meta: {},
+                attributes: {},
+                content: 'b'
+              },
+              value: {
+                element: 'number',
+                meta: {},
+                attributes: {},
+                content: 2
+              }
+            }
+          }
+        ]
+      };
+
+      var element;
+
+      before(function() {
+        element = minim.convertFromEmbedded(obj);
+      });
+
+      it('returns the correct value', function() {
+        expect(element.toRefract()).to.deep.equal(refract);
+      });
+    });
+
+    context('when a property is embedded refract', function() {
+      // Object { a: 1, b: 2, c: 3 } but the value of `b` is embedded refract
+      var obj = {
+        a: 1,
+        b: {
```

Comment:
That is because the other is an object. Because of that, it treats the object as an actual object where the properties are actual properties of the object. 

These other tests cover, what happens when you get a property on a non-object element? We store that data in attributes to preserve that data.